### DISCUSSION TITLE
Commits can be seen in main.js

### DISCUSSION THREAD

**Discussion by KarimKleep** [vercel/community#3802] [Help] [answered]
### Question

I have a project in ReactJs using Vercel to be published.
When I go on my published site, check the Network tab, and go in main.js, I can see the previous commit I have done.

![image](https://github.com/vercel/community/assets/133102466/261acd51-8e8a-4ac2-a82f-1f228c7bcf41)

How can I fix this please ?

**Comment by amyegan** [comment]
Hey @KarimKleep. I'm not sure if you mean that the wrong commit is being deployed, or if you mean something unexpected can be seen when viewing the code in dev tools, or if something else is wrong. Can explain a bit more?

**Reply by KarimKleep** [reply]
I mean when viewing the code in dev tools, in the network tab, we can see informations about the latest commit. Which is considered as sensitive data

**Reply by eden-chan** [reply]
I am also experiencing the same issue.
When I open up the devtools on my Vercel deployed React App, I can see my environment variables in the Sources tab under static/assets/js/main.js

**Reply by wiiiimm** [reply, marked as answer]
@eden-chan @KarimKleep 

React [automatically expose](https://create-react-app.dev/docs/adding-custom-environment-variables/) all ENVs with the prefix `REACT_APP_`.

Vercel expose a set of [system environments variables for different frameworks](https://vercel.com/docs/projects/environment-variables/system-environment-variables#framework-environment-variables) automatically.

You can disable this auto expose feature by following [this step](https://vercel.com/docs/projects/environment-variables/system-environment-variables#).
